### PR TITLE
Add system install test for macOS x86_64

### PR DESCRIPTION
**Pull request by zanieb**
Adds binary builds for x86_64 macOS and a corresponding test

### CODE REVIEW

**Review by charliermarsh — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -364,10 +384,34 @@ jobs:
...
-        run: echo $(which python3.11)
+        run: echo $(which python3)
+
+      - name: "Validate global Python install"
+        run: python3 scripts/check_system_python.py --uv ./uv
+
+  system-test-macos-x86_64:
+    needs: build-binary-macos-x86_64
+    name: "check system | python on macos x86_64"
+    runs-on: macos-14
```
Comment (AlexWaygood) [inline .github/workflows/ci.yml:395]: It looks like the failing job here ran on macOS 12.7.3: https://github.com/AlexWaygood/typeshed-stats/actions/runs/8279567587/job/22654258397?pr=206

Comment (AlexWaygood) [reply .github/workflows/ci.yml:395]: I just use "macos-latest" in that job, but it looks like that does not in fact resolve to the latest macos version

Comment (zanieb) [reply .github/workflows/ci.yml:395]: Oh we should be using `macos-latest` here 🤦

Comment (zanieb) [reply .github/workflows/ci.yml:395]: (I'm a dumbo)

Comment (zanieb) [reply .github/workflows/ci.yml:395]: `macos-14` is the ARM machine

**Pull request merged**